src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -687,4 +691,34 @@
       is_expected.to redirect_to(project_show_path(user.home_project))
     end
   end
+
+  describe 'POST #new_incident_request' do
+    before do
+      login user
+      request.env["HTTP_REFERER"] = root_url # Needed for the redirect_to :back
+    end
+
+    it "without an existent project will raise an exception" do
+      expect { post :new_incident_request, project: 'non:existent:project' }.to raise_error Project::UnknownObjectError
+    end
+
+    context "without a proper action for the maintenance project" do
+      before do
+        post :new_incident_request, project: maintenance_project, description: "Fake description for a request"
+      end
+
+      it { expect(flash[:error]).to eq("MaintenanceHelper::MissingAction") }
+      it { is_expected.to redirect_to(:back) }
+    end
+
+    context "with the proper params" do
+      before do
+        BsRequest.any_instance.stubs(:save!).returns(true)
+        post :new_incident_request, project: maintenance_project, description: "Fake description for a request"
```

Comment:
I can replace stub with expect if think so,... I just wanted to "ignore" the save! call.

You are right, there is no guaranties that the object is saved... because I don't care in this test. I only need to know that if nothing goes wrong it will redirect me to the proper action with the proper flash.

You said "...there is nothing that would ensure that BsRequest get's created and that it has the correct data...", I guess that model tests ensures that, is not the responsibility of a controller spec.

There are many implementations that can match with that test, for sure, and that is one of them. BTW your example won't pass the other test that I added. Someone can change the action to mimic the flow control doing another thing and passing the tests at the same time... but I very doubt it.